// @mtucourses/rate-my-professors/dist/src/queries.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getTeacherQuery = exports.searchTeacherQuery = exports.autocompleteTeacherQuery = exports.autocompleteSchoolQuery = void 0;
const graphql_request_1 = require("graphql-request");
exports.autocompleteSchoolQuery = graphql_request_1.gql `
query AutocompleteSearchQuery(
  $query: String!
)
{
  autocomplete(query: $query) {
    schools {
      edges {
        node {
          id
          name
          city
          state
        }
      }
    }
  }
}`;
exports.autocompleteTeacherQuery = graphql_request_1.gql `
query AutocompleteSearchQuery(
  $query: String!
)
{
  autocomplete(query: $query) {
    teachers {
      edges {
        node {
          id
          firstName
          lastName
          school {
            name
            id
          }
        }
      }
    }
  }
}`;
exports.searchTeacherQuery = graphql_request_1.gql `
query NewSearchTeachersQuery($text: String!, $schoolID: ID!)
{
  newSearch {
    teachers(query: {text: $text, schoolID: $schoolID}) {
      edges {
        cursor
        node {
          id
          firstName
          lastName
          school {
            name
            id
          }
        }
      }
    }
  }
}
`;
exports.getTeacherQuery = graphql_request_1.gql `
query TeacherRatingsPageQuery(
  $id: ID!
) {
  node(id: $id) {
    ... on Teacher {
      id
      firstName
      lastName
      school {
        name
        id
        city
        state
      }
      avgDifficulty
      avgRating
      department
      numRatings
      legacyId
      wouldTakeAgainPercent
    }
    id
  }
}
`;
